classes/ObjectModel.php

```diff
@@ -542,6 +542,8 @@ public function add($auto_date = true, $null_values = false)
             if (count($this->id_shop_list)) {
                 $id_shop_list = $this->id_shop_list;
             }
+        } else {
+            $id_shop_list = Shop::getCompleteListOfShopsID();
```

Comment:
You should actually write this affectation before `if (Shop::isTableAssociated($this->def['table'])) {`.

Because if `(count($this->id_shop_list))` is false, you will have an undefined variable.

Reply:
I agree, changed